launcher/container_runner.go

```diff
@@ -489,27 +489,29 @@ func (r *ContainerRunner) Run(ctx context.Context) error {
 	if err != nil {
 		return &RetryableError{err}
 	}
-	exitStatus, err := task.Wait(ctx)
+	defer task.Delete(ctx)
+
+	exitStatusC, err := task.Wait(ctx)
 	if err != nil {
-		return &RetryableError{err}
+		r.logger.Println(err)
 	}
 	r.logger.Println("workload task started")
 
 	if err := task.Start(ctx); err != nil {
 		return &RetryableError{err}
 	}
-	status := <-exitStatus
+	status := <-exitStatusC
```

Comment:
Try to conform to the example here https://github.com/containerd/containerd/blob/main/docs/getting-started.md